dataplane/dplanerc/interface.go

```diff
@@ -582,118 +615,188 @@ func (ni *Reconciler) reconcile(ctx context.Context, config *oc.Interface) {
 			}
 		}
 	}
+}
 
-	if config.GetOrCreateSubinterface(intf.subintf).Enabled != nil {
-		if state.GetOrCreateSubinterface(intf.subintf).Enabled == nil || config.GetSubinterface(intf.subintf).GetEnabled() != state.GetSubinterface(intf.subintf).GetEnabled() {
-			log.V(1).Infof("setting interface %s enabled %t", data.hostifDevName, config.GetSubinterface(intf.subintf).GetEnabled())
-			if data.hostifID != 0 {
-				_, err := ni.hostifClient.SetHostifAttribute(ctx, &saipb.SetHostifAttributeRequest{
-					Oid:        data.hostifID,
-					OperStatus: proto.Bool(config.GetSubinterface(0).GetEnabled()),
-				})
-				if err != nil {
-					log.Warningf("Failed to set oper status of hostif: %v", err)
-				}
-			}
-			if !data.isAggregate {
-				_, err := ni.portClient.SetPortAttribute(ctx, &saipb.SetPortAttributeRequest{
-					Oid:        data.portID,
-					AdminState: proto.Bool(config.GetSubinterface(0).GetEnabled()),
-				})
-				if err != nil {
-					log.Warningf("Failed to set admin state of port: %v", err)
+func (ni *Reconciler) reconcileSubIntf(ctx context.Context, config, state *oc.Interface) {
+	for idx, subintf := range config.Subinterface {
+		intfRef := ocInterface{
+			name:    config.GetName(),
+			subintf: idx,
+		}
+		data, ok := ni.ocInterfaceData[intfRef]
+		// We have a new interface.
+		if !ok {
+			switch {
+			case config.GetAggregation().GetLagType() != oc.IfAggregate_AggregationType_UNSET:
+				log.Infof("creating new lag interface: %v", intfRefToDevName(intfRef))
+				if err := ni.createLAG(ctx, intfRef, config.GetAggregation().GetLagType()); err != nil {
+					log.Warningf("failed to create lag: %v", err)
 				}
+			case idx != 0 && subintf.Vlan != nil:
```

Comment:
There's nothing in OC that tells you that you can't use subinterface index 0 for a VLAN encapsulation (strictly). We tend not to though (by convention) since there are some platforms that have the idea of an interface w/o a subinterface.

Reply:
is the correct model then that /interfaces/interface == physical port and /interfaces/interface/subinterface == L3 interface

Reply:
or is there a type value i should be using for this. right now i only care about vlan tagged packets getting assigned to different network interface

Reply:
added todo for this